### PR TITLE
Burn Assets

### PR DESCRIPTION
**Pull request by Kais-DkM**

### CODE REVIEW

**Comment by kseo** [comment]
@Kais-DkM Please add burn tests.

File: core/src/parcel.rs [hunk 1]
```diff
@@ -404,12 +404,16 @@ pub struct AssetOutPoint {
...
+pub const UNLOCK_SUCCESS: u8 = 0x01;
+pub const UNLOCK_BURN: u8 = 0x02;
+
 #[derive(Debug, Clone, Eq, PartialEq, RlpDecodable, RlpEncodable, Serialize)]
 #[serde(rename_all = "camelCase")]
 pub struct AssetTransferInput {
     pub prev_out: AssetOutPoint,
     pub lock_script: Bytes,
     pub unlock_script: Bytes,
+    pub unlock_type: u8,
```
Comment (sgkim126) [inline core/src/parcel.rs]: I don't agree with that making this flag.
The most use cases will be just the transfer not burning, and this flag seems verbose.
How about splitting transaction input list?

Comment (Kais-DkM) [reply core/src/parcel.rs]: Agreed. I'll fix this.

File: core/src/state/mod.rs [hunk 2]
```diff
@@ -633,6 +637,7 @@ impl<B: Backend> State<B> {
                 ..
             } => Ok(self.mint_asset(transaction.hash(), metadata, lock_script_hash, parameters, amount, registrar)?),
             Transaction::AssetTransfer {
+                ref burns,
```
Comment (sgkim126) [inline core/src/state/mod.rs]: You can skip `ref` because it's reference binding.

Comment (sgkim126) [reply core/src/state/mod.rs]: I don't think this must be fixed in this patch. It's just for your information.

**Pull request merged**